plugins/inputs/statsd/statsd_test.go

```diff
@@ -125,6 +125,36 @@ func TestCloseConcurrentConns(t *testing.T) {
 }
 
 // benchmark how long it takes to accept & process 100,000 metrics:
+func BenchmarkUDP(b *testing.B) {
+	listener := Statsd{
+		Protocol:               "udp",
+		ServiceAddress:         ":8125",
+		AllowedPendingMessages: 250000,
+	}
+	acc := &testutil.Accumulator{Discard: true}
+
+	// send multiple messages to socket
+	for n := 0; n < b.N; n++ {
+		err := listener.Start(acc)
+		if err != nil {
+			panic(err)
+		}
+
+		time.Sleep(time.Millisecond * 25)
+		conn, err := net.Dial("udp", "127.0.0.1:8125")
```

Comment:
Should we write another benchmark which tests concurrent writing to multiple connections spawned in goroutines ? I think that might be closer to real world scenarios.

Reply:
I'm not sure it will change much in the case of UDP to have multiple senders.  I do wish we had a test that we could run forever, similar to https://github.com/influxdata/influx-stress.  Maybe something as already been written for statsd?